### ISSUE TITLE
Not compatible with Django 2.1

### ISSUE THREAD

**Issue by jepotter1-archive** [kbytesys/django-recaptcha2#18]
My form, which uses `django-crispy-forms` and `django-recaptcha2`, has suddenly stopped working after I updated to 1.0.3 of `django-recaptcha2` and 1.7.2 of `django-crispy-forms`.

Here is the traceback (I've replaced the name of the project and omitted my home directory for privacy reasons):
```
Traceback (most recent call last):
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/core/handlers/base.py", line 124, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "~/Git/django-project/app/views/SignupView.py", line 21, in SignupView
    "form": form
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/shortcuts.py", line 36, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 987, in render
    output = self.filter_expression.resolve(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 698, in resolve
    new_obj = func(obj, *arg_vals)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/crispy_forms/templatetags/crispy_forms_filters.py", line 66, in as_crispy_form
    return template.render(c)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/crispy_forms/templatetags/crispy_forms_utils.py", line 34, in render
    return remove_spaces(self.nodelist.render(context).strip())
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/defaulttags.py", line 209, in render
    nodelist.append(node.render_annotated(context))
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/loader_tags.py", line 188, in render
    return template.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 173, in render
    return self._render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/defaulttags.py", line 309, in render
    return nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/defaulttags.py", line 309, in render
    return nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/defaulttags.py", line 309, in render
    return nodelist.render(context)
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/template/base.py", line 940, in render
    bits.append(str(bit))
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/utils/html.py", line 397, in <lambda>
    klass.__str__ = lambda self: mark_safe(klass_str(self))
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/forms/boundfield.py", line 33, in __str__
    return self.as_widget()
  File "~/.local/share/virtualenvs/django-project-1vOU-KL3/lib/python3.6/site-packages/django/forms/boundfield.py", line 93, in as_widget
    renderer=self.form.renderer,
```
Please let me know if you need any additional information. Thank you! Your work is appreciated!

**Comment by jepotter1-archive** [comment]
Wait... just realized that `pipenv` didn't update either package, it actually only updated the dependencies of the packages. I'm not really sure how Python package management works... do you specify the versions of dependencies or does `pipenv` just grab the latest version out of PyPI?

**Comment by elapouya** [comment]
I have the same errors, but it occured after upgrading from django 2.0 to 2.1
I am using django-recaptcha2 1.0.3 in both cases

pip freeze says : 

```
backcall==0.1.0
certifi==2018.4.16
chardet==3.0.4
decorator==4.3.0
defusedxml==0.5.0
Django==2.1
django-allauth==0.36.0
django-debug-toolbar==1.9.1
django-extensions==2.1.0
django-recaptcha2==1.0.3
idna==2.7
ipython==6.5.0
ipython-genutils==0.2.0
jedi==0.12.1
mysqlclient==1.3.13
oauthlib==2.1.0
parso==0.3.1
pexpect==4.6.0
pickleshare==0.7.4
prompt-toolkit==1.0.15
ptyprocess==0.6.0
Pygments==2.2.0
python3-openid==3.1.0
pytz==2018.5
requests==2.19.1
requests-oauthlib==1.0.0
simplegeneric==0.8.1
six==1.11.0
sqlparse==0.2.4
traitlets==4.3.2
urllib3==1.23
wcwidth==0.1.7
```

If I go back to django 2.0, it works again
How to make it work on Django 2.1 ?

Note : I am using pipenv too.

**Comment by jepotter1-archive** [comment]
Updates:
```
urllib3: 1.22 to 1.23
requests: 2.18.4 to 2.19.1
raven: 2018.4 to 6.9.0
psycopg2-binary: 2.7.4 to 2.7.5
psycopg2: 2.7.4 to 2.7.5
idna: 2.6 to 2.7
gunicorn: 19.8.1 to 19.9.0
django: 2.0.5 to 2.1
beautifulsoup4: 4.6.0 to 4.6.1
```
Looks like the same problem as you @elapouya. Locking Django to `2.0.5` worked!

**Comment by dco5** [comment]
Yeah, same problem with Django 2.1

**Comment by igorcc** [comment]
Just change function render(self, name, value, attrs=None) in the class ReCaptchaWidget
to render(self, name, value, attrs=None, renderer=None):
It works like a charm!

**Comment by kbytesys** [comment]
this issue was fixed by https://github.com/kbytesys/django-recaptcha2/pull/19 already merged and published in the 1.1.0 release. Bye!
